src/BridgedRoom.ts

```diff
@@ -206,10 +224,32 @@ export class BridgedRoom {
         this.slackBotId =  opts.slack_bot_id;
         this.accessToken = opts.access_token;
         this.accessScopes = opts.access_scopes;
+        this.slackType = opts.slack_type || "channel";
+        if (opts.is_private === undefined) {
+            opts.is_private = false;
+        }
+        this.isPrivate = opts.is_private;
+        this.puppetOwner = opts.puppet_owner;
 
         this.dirty = true;
     }
 
+    public updateUsingChannelInfo(channelInfo: ConversationsInfoResponse) {
+        const chan = channelInfo.channel;
+        this.setValue("isPrivate", chan.is_private);
+        if (chan.is_channel) {
+            this.setValue("slackType", "channel");
+        } else if (chan.is_group) {
+            this.setValue("slackType", "group");
+        } else if (chan.is_mpim) {
```

Comment:
looking at this, I think this is the wrong way around. `is_group` seems to be true whenever `is_mpim` is true